go.mod

```diff
@@ -10,29 +10,34 @@ require (
 	github.com/gorilla/mux v1.8.0
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0
 	github.com/spf13/cast v1.5.0
-	github.com/spf13/cobra v1.4.0
+	github.com/spf13/cobra v1.5.0
 	github.com/stretchr/testify v1.8.0
-	github.com/tendermint/tendermint v0.35.4
+	github.com/tendermint/tendermint v0.35.2
 	github.com/tendermint/tm-db v0.6.7
-	golang.org/x/crypto v0.0.0-20220411220226-7b82a4e95df4 // indirect
-	golang.org/x/net v0.0.0-20220520000938-2e3eb7b945c2 // indirect
-	golang.org/x/sys v0.0.0-20220520151302-bc2c85ada10a // indirect
-	golang.org/x/term v0.0.0-20210927222741-03fcf44c2211 // indirect
-	google.golang.org/genproto v0.0.0-20220519153652-3a47de7e79bd
-	google.golang.org/grpc v1.47.0
+	golang.org/x/crypto v0.0.0-20220525230936-793ad666bf5e // indirect
+	golang.org/x/net v0.0.0-20220617184016-355a448f1bc9 // indirect
+	golang.org/x/sys v0.0.0-20220702020025-31831981b65f // indirect
+	golang.org/x/term v0.0.0-20220526004731-065cf7ba2467 // indirect
+	google.golang.org/genproto v0.0.0-20220617124728-180714bec0ad
```

Comment:
[no change needed] I was curious about why there are three require blocks in this go.mod file and it seems like these `// indirect` deps can be moved to the last `require` block if we run `go mod tidy -go=1.17`. Separating direct deps from transitive deps is visual convenience so no change needed.

See: https://stackoverflow.com/questions/69432877/why-there-are-two-require-blocks-in-go-mod-since-go-1-17/69433550

I'm still unsure on why the second require block which contains direct dependencies is distinct from the first one. Perhaps because it includes `-beta` versions?

Reply:
I can confirm that running `go mod tidy` locally makes more changes to `go.mod` and `go.sum`.
@rootulp Btw, why do you need  to run `go mod tidy -go=1.17`  ? I guess  `go mod tidy` is enough no?

Reply:
> Btw, why do you need to run `go mod tidy -go=1.17`

I didn't try this on the PR (yet) but this was based on the StackOverflow post which says:

> and/or run go mod tidy -go=1.17

More docs:
```shell
go help mod tidy
...
The -go flag causes tidy to update the 'go' directive in the go.mod
file to the given version, which may change which module dependencies
are retained as explicit requirements in the go.mod file.
(Go versions 1.17 and higher retain more requirements in order to
support lazy module loading.)
```

so it seems like Go 1.17+ will format `go.mod` slightly differently. Given the Go version installed on our machines + in this package's `go.mod` is already Go 1.17, the `-go=1.17` flag likely isn't necessary so apologies for the confusion

Reply:
I think the repo is currently using `1.18`

Reply:
You're right b/c https://github.com/celestiaorg/celestia-app/pull/573
Based on my understanding of the above docs, 1.17+ will format go.mod `require` sections in separate blocks.